src/lib/classes/styling/GlobalCssSchemesLoader.ts

```diff
@@ -0,0 +1,206 @@
+import type ColorValue from "./ZodSchemas/AttributeSchemas/ColorAttribute";
+import type MediaScheme from "./ZodSchemas/MediaScheme";
+
+import MediaSchemes from "./ZodSchemas/MediaSchemes";
+import GlobalCssProperties from "../../GlobalCssProperties.json";
+
+import type { z } from "zod";
+
+/**
+ * Class for handling the loading of different properties based on active media features
+ */
+class GlobalCssSchemesLoader {
+	private _window: Window;
+	private _mediaSchemes: z.infer<typeof MediaSchemes>;
+	private _propertyNames: string[] = [];
+
+	// SUPPORTED MEDIA FEATURES
+	private _supportedMediaFeatures: string[] = [
+		"prefers-color-scheme: dark",
+		"prefers-color-scheme: light",
+		"prefers-reduced-motion",
+		"prefers-reduced-transparency",
+	];
```

Comment:
Why are we limiting the media features that one can use? The browser knows which media features it supports and will just return `false` if it doesn't support the media feature.

Reply:
I can see now that you are using it to define which media features we are listening on. I think it would be better if we only listened on the media features that the theme is actually using.

If this is a major rewrite, you don't have to do this, we can look at it in a future PR, maybe make an issue for it.